src/content/blog/how-to-configure-astropaper-theme.md
---
author: Sat Naing
pubDatetime: 2022-09-23T04:58:53Z
title: How to configure AstroPaper theme
postSlug: how-to-configure-astropaper-theme
featured: true
draft: false
tags:
  - configuration
  - docs
description: How you can make AstroPaper theme absolutely yours.
---

AstroPaper is a highly customizable Astro blog theme. With AstroPaper, you can customize everything according to your personal taste. This article will explain how you can make some customizations easily in the config file.

## Table of contents

## Configuring SITE

The important configurations lies in `src/config.ts` file. Within that file, you'll see the `SITE` object where you can specify your website's main configurations.

During deveopment, it's okay to leave `SITE.website` empty. But in production mode, you should specify your deployed url in `SITE.website` option since this will be used for canonical URL, social card URL etc.. which are important for SEO.

```js
// file: src/config.ts
export const SITE = {
  website: 'https://astro-paper.pages.dev/',
  author: 'Sat Naing',
  desc: 'A minimal, responsive and SEO-friendly Astro blog theme.',
  title: 'AstroPaper',
  ogImage: 'og-low-res.jpeg',
  lightAndDarkMode: true,
  postPerPage: 3
};
```

Here are SITE configuration options

| Options            | Description                                                                                                                                                  |
| ------------------ | ------------------------------------------------------------------------------------------------------------------------------------------------------------ |
| `website`          | Your deployed website url                                                                                                                                    |
| `author`           | Your name                                                                                                                                                    |
| `desc`             | Your site description. Useful for SEO and social media sharing.                                                                                              |
| `title`            | Your site name                                                                                                                                               |
| `ogImage`          | Your default OG image for the site. Useful for social media sharing. OG images can be an external image url or they can be placed under `/public` directory. |
| `lightAndDarkMode` | Enable or disable `light & dark mode` for the website. If disabled, primary color scheme will be used. This option is enabled by default.                    |
| `postPerPage`      | You can specify how many posts will be displayed in each posts page. (eg: if you set SITE.postPerPage to 3, each page will only show 3 posts per page)       |

## Configuring locale

You can configure the default locale used for the build (e.g., date format in the post page), and for the rendering in browsers (e.g., date format in the search page)

```js
// file: src/config.ts
export const LOCALE = ['en-EN']; // set to [] to use the environment default
```

You can even specify an array of locales for fallback languages. Leave it empty `[]` to use the environment default at _build-_ and _run-time_.

## Configuring logo or title

You can specify site's title or logo image in `src/config.ts` file.

![An arrow pointing at the website logo](https://res.cloudinary.com/noezectz/v1663911318/astro-paper/AstroPaper-logo-config_goff5l.png)

```js
// file: src/config.ts
export const LOGO_IMAGE = {
  enable: false,
  svg: true,
  width: 216,
  height: 46
};
```

If you specify `LOGO_IMAGE.enable` => `false`, AstroPaper will automatically convert `SITE.title` to the main site text logo.

If you specify `LOGO_IMAGE.enable` => `true`, AstroPaper will use the logo image as the site's main logo.

You have to specify `logo.png` or `logo.svg` under `/public/assets` directory. Currently, only svg and png image file formats are supported. (**_Important!_** _logo name has to be logo.png or logo.svg)_

If your logo image is png file format, you have to set `LOGO_IMAGE.svg` => `false`.

It is recommended that you specify width and height of your logo image. You can do that by setting `LOGO_IMAGE.width` _and_ `LOGO_IMAGE.height`

## Configuring social links

You can configure your own social links along with its icons.

![An arrow pointing at social link icons](https://res.cloudinary.com/noezectz/v1663914759/astro-paper/astro-paper-socials_tkcjgq.png)

Currently 20 social icons are supported. (Github, LinkedIn, Facebook etc.)

You can specify and enable certain social links in hero section and footer. To do this, go to `/src/config.ts` and then you'll find `SOCIALS` array of object.

```js
// file: src/config.ts
export const SOCIALS: SocialObjects = [
  {
    name: "Github",
    href: "https://github.com/satnaing/astro-paper",
    linkTitle: ` ${SITE.title} on Github`,
    active: true,
  },
  {
    name: "Facebook",
    href: "https://github.com/satnaing/astro-paper",
    linkTitle: `${SITE.title} on Facebook`,
    active: true,
  },
  {
    name: "Instagram",
    href: "https://github.com/satnaing/astro-paper",
    linkTitle: `${SITE.title} on Instagram`,
    active: true,
  },
  ...
]
```

You have to set specific social link to `active: true` in order to appear your social links in hero and footer section. Then, you also have to specify your social link in `href` property.

For instance, if I want to make my Github appear, I'll make it like this.

```js
export const SOCIALS: SocialObjects = [
  {
    name: "Github",
    href: "https://github.com/satnaing", // update account link
    linkTitle: `${SITE.title} on Github`, // this text will appear on hover and VoiceOver
    active: true, // makre sure to set active to true
  }
  ...
]
```

Another thing to note is that you can specify the `linkTitle` in the object. This text will display when hovering on the social icon link. Besides, this will improve accessibility and SEO. AstroPaper provides default link title values; but you can replace them with your own texts.

For example,

```js
linkTitle: `${SITE.title} on Twitter`,
```

to

```js
linkTitle: `Follow ${SITE.title} on Twitter`;
```

## Conclusion

This is the brief specification of how you can customize this theme. You can customize more if you know some coding. For customizing styles, please read [this article](https://astro-paper.pages.dev/posts/customizing-astropaper-theme-color-schemes/). Thanks for reading.✌🏻


CHANGELOG.md
# Changelog

## [1.4.1](https://github.com/yamcodes/yam.codes/compare/v1.4.0...v1.4.1) (2023-10-13)


### Bug Fixes

* make og description shorter ([30a1289](https://github.com/yamcodes/yam.codes/commit/30a128900c5a1f21a2a9804a64d4a25e0d9f8e83))

## [1.4.0](https://github.com/yamcodes/yam.codes/compare/v1.3.0...v1.4.0) (2023-10-13)


### Features

* add open graph description ([5931c7e](https://github.com/yamcodes/yam.codes/commit/5931c7ee6f8fc80d6f00f289950c1353aed44e03))


### Bug Fixes

* astro formatting ([203126b](https://github.com/yamcodes/yam.codes/commit/203126b2b46392b8821b635515d453d428d4e834))

## [1.3.0](https://github.com/yamcodes/yam.codes/compare/v1.2.0...v1.3.0) (2023-10-13)


### Features

* format markdown in lint-staged ([1078afa](https://github.com/yamcodes/yam.codes/commit/1078afa0f265340a0559714702b89767eac678d4))
* update og picture ([d43f213](https://github.com/yamcodes/yam.codes/commit/d43f2136fd49772c11b415b8898373c6449dacd0))


### Bug Fixes

* lint files ([1b6d20d](https://github.com/yamcodes/yam.codes/commit/1b6d20d56ecd5fffb7c072c950ff13d85bfb1e4f))

## [1.2.0](https://github.com/yamcodes/yam.codes/compare/v1.1.0...v1.2.0) (2023-10-13)


### Features

* clean up README ([7a16d35](https://github.com/yamcodes/yam.codes/commit/7a16d35b328a8985b7b987fa7728f9341c1cba88))
* rename astropaper to yamcodes ([c3fe684](https://github.com/yamcodes/yam.codes/commit/c3fe684e0f2bc7688477bfe4b4df8f1922fce7d9))


### Bug Fixes

* remove deprecated args from lint-staged ([c726a97](https://github.com/yamcodes/yam.codes/commit/c726a972f84f37be06428170ece35823570fed11))

## [1.1.0](https://github.com/yamcodes/yam.codes/compare/v1.0.0...v1.1.0) (2023-10-13)


### Features

* lint js in markdown ([550b8c8](https://github.com/yamcodes/yam.codes/commit/550b8c8d44ca7ac98743bf62cb7c6eda26417aed))


### Bug Fixes

* clean up CHANGELOG ([2cdfd9b](https://github.com/yamcodes/yam.codes/commit/2cdfd9b6b3e0aeda6b6862561d4d93da522850d5))
* stop linting markdown ([450d3eb](https://github.com/yamcodes/yam.codes/commit/450d3eb60ea666da9c8f1555718289c605b4b0ee))

## 1.0.0 (2023-10-13)


### Features

* astro vercel adapter ([61a8b6f](https://github.com/yamcodes/yam.codes/commit/61a8b6ff26e4f0aca85f10201c75ee045d3fd621))
* initialize playwright ([5f4f820](https://github.com/yamcodes/yam.codes/commit/5f4f820e01bd34359d442e8141cd6a722844e30e))
* release-please ([ce7da39](https://github.com/yamcodes/yam.codes/commit/ce7da39e7cd188922ee1dfb161f7ae48da7d3c4a))
* swpm ([cb306f8](https://github.com/yamcodes/yam.codes/commit/cb306f8806a43d8e8146e2aa757d71bdd3f105e6))
* use squoosh for images ([e3c958c](https://github.com/yamcodes/yam.codes/commit/e3c958c312bf2f1a1f1d8eca0c5f5a0e2aeab2b5))
* use vercel static adapter ([b0e21a4](https://github.com/yamcodes/yam.codes/commit/b0e21a4fba9c7f2ff1df572034cdb69d386cdfc2))
* vercel adapter ([23a3f64](https://github.com/yamcodes/yam.codes/commit/23a3f6437c1ddd4b72208c36d0213cbb64bd49b5))


### Bug Fixes

* build ([18d18fa](https://github.com/yamcodes/yam.codes/commit/18d18fa5bc79270edd6a56d9d377d9ec85a54a3b))
* build ([aefa790](https://github.com/yamcodes/yam.codes/commit/aefa790409ba05574e717ca1f1ee4edc20762e94))
* commit hooks ([d19d9fa](https://github.com/yamcodes/yam.codes/commit/d19d9faecaefa9a77eb37654040db835f70ab2c2))
* disable jampack ([d89bca9](https://github.com/yamcodes/yam.codes/commit/d89bca93598025f8fe82a80d76728dab44408533))
* format and lint ([971de82](https://github.com/yamcodes/yam.codes/commit/971de82279ab6e6f9419807541184212683f3c1e))
* formatting issues ([44afa58](https://github.com/yamcodes/yam.codes/commit/44afa586d239b5978d25db3c2e52dd9bb117c48c))
* images not displaying ([6bbcdea](https://github.com/yamcodes/yam.codes/commit/6bbcdea2ab69a12f756f4cde97820150b9120982))
* lint ([11af768](https://github.com/yamcodes/yam.codes/commit/11af7683aecd5f576f40568568d209ea97ed419b))
* lint issue ([d743217](https://github.com/yamcodes/yam.codes/commit/d743217603e7ccde7b65b5cda38fe41ed45d82e0))
* lint warning ([4f8164c](https://github.com/yamcodes/yam.codes/commit/4f8164caae85f282100e56b35418fa3e4bcc14d4))
* lint-staged ([64c18f9](https://github.com/yamcodes/yam.codes/commit/64c18f9df41d6ba1d379807cd5bf442ca5287a2c))
* pages not being rendered ([e0d574d](https://github.com/yamcodes/yam.codes/commit/e0d574dc2b7da8abd58c3ca0fd5347b4c21e0546))
* stop using squoosh image service ([db8dbfc](https://github.com/yamcodes/yam.codes/commit/db8dbfcdb5d2254b7f6af1b25b4a53c61cd7b7ff))
* temporarily disable tests ([e963b71](https://github.com/yamcodes/yam.codes/commit/e963b71fd2f932dd7cfcdfcafc2a0e9a7286e050))
* tests ([49fabfd](https://github.com/yamcodes/yam.codes/commit/49fabfd954e1df3d9cdb735b7bf15b0c1bffe89e))
* vercel build ([a9f8730](https://github.com/yamcodes/yam.codes/commit/a9f87301b16125dd27fb36d2eb04d90911481c34))


src/config.ts
import type { Site, SocialObjects } from './types';

export const SITE: Site = {
  website: 'https://yam.codes/', // replace this with your deployed domain
  author: 'yamcodes',
  desc: 'Developer. Designer. Open-source enthusiast.',
  title: 'Yam Borodetsky',
  ogImage: 'og-low-res.jpg',
  lightAndDarkMode: true,
  postPerPage: 3
};

export const LOCALE = ['en-EN']; // set to [] to use the environment default

export const LOGO_IMAGE = {
  enable: false,
  svg: true,
  width: 216,
  height: 46
};

export const SOCIALS: SocialObjects = [
  {
    name: 'Github',
    href: 'https://github.com/satnaing/astro-paper',
    linkTitle: ` ${SITE.title} on Github`,
    active: true
  },
  {
    name: 'Facebook',
    href: 'https://github.com/satnaing/astro-paper',
    linkTitle: `${SITE.title} on Facebook`,
    active: true
  },
  {
    name: 'Instagram',
    href: 'https://github.com/satnaing/astro-paper',
    linkTitle: `${SITE.title} on Instagram`,
    active: true
  },
  {
    name: 'LinkedIn',
    href: 'https://github.com/satnaing/astro-paper',
    linkTitle: `${SITE.title} on LinkedIn`,
    active: true
  },
  {
    name: 'Mail',
    href: 'mailto:yourmail@gmail.com',
    linkTitle: `Send an email to ${SITE.title}`,
    active: false
  },
  {
    name: 'Twitter',
    href: 'https://github.com/satnaing/astro-paper',
    linkTitle: `${SITE.title} on Twitter`,
    active: false
  },
  {
    name: 'Twitch',
    href: 'https://github.com/satnaing/astro-paper',
    linkTitle: `${SITE.title} on Twitch`,
    active: false
  },
  {
    name: 'YouTube',
    href: 'https://github.com/satnaing/astro-paper',
    linkTitle: `${SITE.title} on YouTube`,
    active: false
  },
  {
    name: 'WhatsApp',
    href: 'https://github.com/satnaing/astro-paper',
    linkTitle: `${SITE.title} on WhatsApp`,
    active: false
  },
  {
    name: 'Snapchat',
    href: 'https://github.com/satnaing/astro-paper',
    linkTitle: `${SITE.title} on Snapchat`,
    active: false
  },
  {
    name: 'Pinterest',
    href: 'https://github.com/satnaing/astro-paper',
    linkTitle: `${SITE.title} on Pinterest`,
    active: false
  },
  {
    name: 'TikTok',
    href: 'https://github.com/satnaing/astro-paper',
    linkTitle: `${SITE.title} on TikTok`,
    active: false
  },
  {
    name: 'CodePen',
    href: 'https://github.com/satnaing/astro-paper',
    linkTitle: `${SITE.title} on CodePen`,
    active: false
  },
  {
    name: 'Discord',
    href: 'https://github.com/satnaing/astro-paper',
    linkTitle: `${SITE.title} on Discord`,
    active: false
  },
  {
    name: 'GitLab',
    href: 'https://github.com/satnaing/astro-paper',
    linkTitle: `${SITE.title} on GitLab`,
    active: false
  },
  {
    name: 'Reddit',
    href: 'https://github.com/satnaing/astro-paper',
    linkTitle: `${SITE.title} on Reddit`,
    active: false
  },
  {
    name: 'Skype',
    href: 'https://github.com/satnaing/astro-paper',
    linkTitle: `${SITE.title} on Skype`,
    active: false
  },
  {
    name: 'Steam',
    href: 'https://github.com/satnaing/astro-paper',
    linkTitle: `${SITE.title} on Steam`,
    active: false
  },
  {
    name: 'Telegram',
    href: 'https://github.com/satnaing/astro-paper',
    linkTitle: `${SITE.title} on Telegram`,
    active: false
  },
  {
    name: 'Mastodon',
    href: 'https://github.com/satnaing/astro-paper',
    linkTitle: `${SITE.title} on Mastodon`,
    active: false
  }
];
